src/app/modules/attachment/attachment.service.ts

```diff
@@ -0,0 +1,51 @@
+import crypto from 'crypto'
+import { ResultAsync } from 'neverthrow'
+
+import { aws as AwsConfig } from '../../../config/config'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { AttachmentUploadError } from '../core/core.errors'
+
+const logger = createLoggerWithLabel(module)
+
+export const uploadAttachments = (
+  formId: string,
+  attachmentData: Record<string, unknown>,
+): ResultAsync<Map<string, string>, AttachmentUploadError> => {
```

Comment:
perhaps it would be clearer to define a type alias `type AttachmentMetadata = Map<string, string>` in `encrypt-submission.types, and also to reduce repetition